coreneuron/io/reports/report_configuration_parser.cpp

```diff
@@ -40,13 +40,34 @@ enum class TargetType {
 };
 
 /*
- * Split filter string ("mech.var_name") into mech_id and var_name
+ * Split filter comma separated strings ("mech.var_name") into mech_name and var_name
  */
 void parse_filter_string(const std::string& filter, ReportConfiguration& config) {
-    std::istringstream iss(filter);
-    std::string token;
-    std::getline(iss, config.mech_name, '.');
-    std::getline(iss, config.var_name, '.');
+    std::istringstream in(filter);
+
+    std::vector<std::string> mechanisms;
+    std::stringstream ss(filter);
+    std::string mechanism;
+    // Separate in case of more than 1 report_on
+    while (getline(ss, mechanism, ',')) {
+        mechanisms.push_back(mechanism);
+    }
+    // Split mechanisms
+    for (int i = 0; i < mechanisms.size(); ++i) {
+        std::string mech_name;
+        std::string var_name;
+        std::istringstream iss(mechanisms[i]);
+        std::getline(iss, mech_name, '.');
+        std::getline(iss, var_name, '.');
+        if (var_name.empty()) {
+            var_name = "i";
+        }
```

Comment:
@jorblancoa : I assume this require changes in neurodamus as well and hence you want to avoid it doing another iteration?

Reply:
It could, but I would need to add quite a bit of extra functionality in neurodamus (check if summation report and then split report_on and see if there is variable name...). Right now this is done in neurodamus after calling the addSummationReport (we skip in case of coreneuron) in the method evaluateSummationReportOn() so we would need to repeat all the logic while in coreneuron is just 2 lines of code.